flood_adapt/object_model/hazard/event/event.py

```diff
@@ -99,21 +100,69 @@ def timeseries_shape(
 
     @staticmethod
     def read_csv(csvpath: Union[str, Path]) -> pd.DataFrame:
-        """Read a timeseries file and return a pd.Dataframe.
+        """Read a timeseries file and return a pd.DataFrame.
 
         Parameters
         ----------
         csvpath : Union[str, os.PathLike]
-            path to csv file
+            Path to the CSV file.
 
         Returns
         -------
         pd.DataFrame
-            Dataframe with time as index and waterlevel as first column.
+            Dataframe with time as index and waterlevel as the first column.
         """
-        df = pd.read_csv(csvpath, index_col=0, header=None)
+        num_columns = None
+        has_header = None
+
+        with open(csvpath, "r") as f:
+            try:
+                # read the first 1024 bytes to determine if there is a header
+                has_header = csv.Sniffer().has_header(f.read(1024))
+            except csv.Error:
+                # The file is empty
+                has_header = False
+
+            f.seek(0)
+            reader = csv.reader(f, delimiter=",")
+            num_columns = len(next(reader)) - 1  # subtract 1 for the index column
+
+        if has_header is None:
+            raise ValueError(
+                f"Could not determine if the CSV file has a header: {csvpath}."
+            )
+        header = 0 if has_header else None
+        df = pd.read_csv(csvpath, index_col=0, header=header)
+
```

Comment:
Reading a CSV file doesn't actually change whether you have a header or not, so I think it's much easier to just read in the csv file with pandas as it it doesn't have a header, do your checks and set the first row as the header if you've decided there is one.

Reply:
First of all, reading a file does change whether there are headers or not. if there is no header, it uses the first value in the data column as the name of the column.
Second, regarding your suggestion, that is essentially what Im doing, I check if it has headers, and then read it accordingly? I could not find a way in pandas that returns if there is a header or not. Do you have a suggestion?

Reply:
I'll make a suggestion one minute